accounts/accounts.go

```diff
@@ -22,12 +42,69 @@ func NewAccountStore() *AccountStore {
 	return &as
 }
 
+func (ac Account) addStockToPortfolio(stock string, units int) bool {
+	currentUnits, ok := ac.portfolio[stock]
+	if !ok {
+		currentUnits = 0
+	}
+	ac.portfolio[stock] = currentUnits + units
+	return true
+}
+
+func (ac Account) removeStockFromPortfolio(stock string, units int) bool {
+	currentUnits, ok := ac.portfolio[stock]
+	if !ok || currentUnits - units < 0{
+		log.Notice("User does not have enough stock to sell")
+		return false
+	}
+	ac.portfolio[stock] = currentUnits - units
+	return true
+}
+
+func (ac Account) getPortfolioStockUnits(stock string) int {
+	return ac.portfolio[stock]
+
+}
+
 // HasAccount : Checks if there's an existing account for the user
 func (as AccountStore) HasAccount(name string) bool {
 	_, ok := as.Accounts[name]
 	return ok
 }
 
+// GetAccount ; Grab an account if it exists for the user
+func (as AccountStore) GetAccount(name string) *Account {
+	account, ok := as.Accounts[name]
+	if !ok {
+		return nil;
+	}
+	return account
+}
+
+// AddToBuyQueue ; Add a stock S to the buy queue
+func (ac Account) AddToBuyQueue(stock string, units uint, unitPrice currency.Currency) bool {
```

Comment:
Well, I think it should return false at some point. We're going to inevitably fail to add to the buy queue for some reason.